calyx-py/test/correctness/queues/binheap/fifo.py

```diff
@@ -0,0 +1,53 @@
+# pylint: disable=import-error
+import sys
+import calyx.builder as cb
+import calyx.queue_call as qc
+from stable_binheap import insert_stable_binheap
+
+
+def insert_binheap_fifo(prog, name, queue_size_factor):
+    """Inserts the component `fifo` into the program.
+
+    It is a first in, first out queue implemented via binary heap
+
+    It has:
+    - two inputs, `cmd` and `value`.
+    - one memory, `mem`, of size `2**queue_size_factor`.
+    - two ref registers, `ans` and `err`.
+    """
+    comp = prog.component(name)
+
+    binheap = insert_stable_binheap(prog, "binheap", queue_size_factor)
+    binheap = comp.cell("binheap", binheap)
+
+    cmd = comp.input("cmd", 2)
+    value = comp.input("value", 32)
+
+    ans = comp.reg(32, "ans", is_ref=True)
+    err = comp.reg(1, "err", is_ref=True)
+
+    comp.control += [
+        cb.invoke(
+            binheap,
+            in_value=value,
+            in_rank=cb.const(32, 1),
```

Comment:
Consider a comment here or in the doctstring explaining the strategy. You are sorta getting away with a really simple scheduling transaction (rank `1` to everyone) and it may warrant a little note!

Reply:
Done in [de13dc7](https://github.com/calyxir/calyx/pull/2164/commits/de13dc7e0ea656d8e452d19f98651e75c05c13f9)